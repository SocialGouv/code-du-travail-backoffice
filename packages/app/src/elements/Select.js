import styled from "@emotion/styled";
import PropTypes from "prop-types";
import React from "react";
import ReactSelect from "react-select";

const StyledReactSelect = styled(ReactSelect)`
  font-size: 0.875rem;
  margin-left: ${p => (p.withMarginLeft ? "1rem" : 0)};
  margin-right: ${p => (p.withMarginRight ? "1rem" : 0)};
  width: 15rem;

  .class__control {
    border-color: var(--color-border);
    border-radius: 0;
    min-height: 0;
  }

  .class__value-container {
    max-height: 6rem;
    padding: 0.3rem 0.4rem;
    overflow-y: auto;
  }

  .class__dropown-indicator {
    padding: 0.325rem;
  }

  .class__indicator-separator {
    margin: 0.325rem 0;
  }

  .class__input {
    padding: 0;
  }

  .class__placeholder {
    color: var(--color-placeholder);
  }

  .class__multi-value {
    border-radius: 0;
  }

  .class__menu {
    border: solid 1px var(--color-border);
    border-radius: 0;
    border-top: 0;
    box-shadow: none;
    margin: 0;
  }

  .class__menu-list {
    padding: 0;
  }
`;

/**
 * Be careful! The {instanceId} prop is mandatory to avoid "Prop `id` did not match." warning.
 *
 * @see https://github.com/trezor/trezor-suite/issues/290#issuecomment-516349580
 */
const Select = React.forwardRef(
  ({ withMarginLeft = false, withMarginRight = false, ...props }, ref) => (
    <StyledReactSelect
      classNamePrefix="class"
      ref={ref}
      withMarginLeft={withMarginLeft}
      withMarginRight={withMarginRight}
      {...props}
    />
  ),
);

Select.propTypes = {
  clearValue: PropTypes.func,
  emotion: PropTypes.any,
  getStyles: PropTypes.func,
  getValue: PropTypes.func,
  hasValue: PropTypes.bool,
  instanceId: PropTypes.string.isRequired,
  isMulti: PropTypes.bool,
  options: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)),
  selectOption: PropTypes.func,
  selectProps: PropTypes.any,
  setValue: PropTypes.func,
  withMarginLeft: PropTypes.bool,
  withMarginRight: PropTypes.bool,
};

Select.displayName = "Select";

export default Select;
