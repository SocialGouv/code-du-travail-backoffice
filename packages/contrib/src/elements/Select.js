import React from "react";
import ReactSelect from "react-select";

const styles = {
  container: s => ({
    ...s,
    fontSize: "0.875rem",
    width: "15rem"
  }),
  control: s => ({
    ...s,
    borderRadius: 0,
    minHeight: 0
  }),
  dropdownIndicator: s => ({
    ...s,
    padding: "0.325rem"
  }),
  indicatorSeparator: s => ({
    ...s,
    margin: "0.325rem 0"
  }),
  input: s => ({
    ...s,
    padding: 0
  }),
  menu: s => ({
    ...s,
    border: "solid 1px var(--color-border)",
    borderTop: 0,
    borderRadius: 0,
    boxShadow: "none",
    margin: 0
  }),
  menuList: s => ({
    ...s,
    padding: 0
  })
};

const Select = (props, ref) => <ReactSelect ref={ref} styles={styles} {...props} />;

export default React.forwardRef(Select);
