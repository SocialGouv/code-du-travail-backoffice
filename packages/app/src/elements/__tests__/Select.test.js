import React from "react";

import Select from "../Select";

describe("elements/<Select />", () => {
  it(`should pass`, () => {
    const βselect = testRender(<Select instanceId="an-instance-id" />);

    expect(βselect).toMatchSnapshot();
  });

  it(`should pass with {withMarginLeft}`, () => {
    const βselect = testRender(<Select instanceId="an-instance-id" withMarginLeft />);

    expect(βselect).toHaveStyleRule("margin-left", "1rem");
    expect(βselect).toHaveStyleRule("margin-right", "0");
  });

  it(`should pass with {withMarginRight}`, () => {
    const βselect = testRender(<Select instanceId="an-instance-id" withMarginRight />);

    expect(βselect).toHaveStyleRule("margin-left", "0");
    expect(βselect).toHaveStyleRule("margin-right", "1rem");
  });
});
