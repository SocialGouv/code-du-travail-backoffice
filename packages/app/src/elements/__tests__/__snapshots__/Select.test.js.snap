// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/<Select /> should pass 1`] = `
.emotion-9 {
  position: relative;
  box-sizing: border-box;
  font-size: 0.875rem;
  margin-left: 0;
  margin-right: 0;
  width: 15rem;
}

.emotion-9 .class__control {
  border-color: var(--color-border);
  border-radius: 0;
  min-height: 0;
}

.emotion-9 .class__value-container {
  max-height: 6rem;
  padding: 0.3rem 0.4rem;
  overflow-y: auto;
}

.emotion-9 .class__dropown-indicator {
  padding: 0.325rem;
}

.emotion-9 .class__indicator-separator {
  margin: 0.325rem 0;
}

.emotion-9 .class__input {
  padding: 0;
}

.emotion-9 .class__placeholder {
  color: var(--color-placeholder);
}

.emotion-9 .class__multi-value {
  border-radius: 0;
}

.emotion-9 .class__menu {
  border: solid 1px var(--color-border);
  border-radius: 0;
  border-top: 0;
  box-shadow: none;
  margin: 0;
}

.emotion-9 .class__menu-list {
  padding: 0;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(0,0%,100%);
  border-color: hsl(0,0%,80%);
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 38px;
  outline: 0 !important;
  position: relative;
  -webkit-transition: all 100ms;
  transition: all 100ms;
  box-sizing: border-box;
}

.emotion-7:hover {
  border-color: hsl(0,0%,70%);
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 8px;
  -webkit-overflow-scrolling: touch;
  position: relative;
  overflow: hidden;
  box-sizing: border-box;
}

.emotion-0 {
  color: hsl(0,0%,50%);
  margin-left: 2px;
  margin-right: 2px;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  box-sizing: border-box;
}

.emotion-1 {
  margin: 2px;
  padding-bottom: 2px;
  padding-top: 2px;
  visibility: visible;
  color: hsl(0,0%,20%);
  box-sizing: border-box;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-3 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: hsl(0,0%,80%);
  margin-bottom: 8px;
  margin-top: 8px;
  width: 1px;
  box-sizing: border-box;
}

.emotion-5 {
  color: hsl(0,0%,80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-5:hover {
  color: hsl(0,0%,60%);
}

.emotion-4 {
  display: inline-block;
  fill: currentColor;
  line-height: 1;
  stroke: currentColor;
  stroke-width: 0;
}

<div
  className="emotion-8 emotion-9"
  onKeyDown={[Function]}
>
  <div
    className="class__control emotion-7"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="class__value-container emotion-2"
    >
      <div
        className="class__placeholder emotion-0"
      >
        Select...
      </div>
      <div
        className="emotion-1"
      >
        <div
          className="class__input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-an-instance-id-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "label": "input",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="class__indicators emotion-6"
    >
      <span
        className="class__indicator-separator emotion-3"
      />
      <div
        aria-hidden="true"
        className="class__indicator class__dropdown-indicator emotion-5"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="emotion-4"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
